datasource/image/data.go

```diff
@@ -0,0 +1,191 @@
+package image
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type DatasourceOutput,Config
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/packer-plugin-sdk/common"
+	"github.com/hashicorp/packer-plugin-sdk/hcl2helper"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/config"
+	"github.com/linode/linodego"
+	"github.com/linode/packer-plugin-linode/helper"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type Datasource struct {
+	config Config
+}
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	helper.LinodeCommon `mapstructure:",squash"`
+
+	// Matching the label of an image by exact label
+	Label string `mapstructure:"label"`
+
+	// Matching the label of an image by a regular expression
+	LabelRegex string `mapstructure:"label_regex"`
+
+	// Matching the ID of an image by exact ID
+	ID string `mapstructure:"id"`
+
+	// Matching the ID of an image by a regular expression
+	IDRegex string `mapstructure:"id_regex"`
+
+	// Whether to use the latest created image when there are multiple matches
+	Latest bool `mapstructure:"latest"`
+}
+
+func (d *Datasource) ConfigSpec() hcldec.ObjectSpec {
+	return d.config.FlatMapstructure().HCL2Spec()
+}
+
+func (d *Datasource) Configure(raws ...interface{}) error {
+	err := config.Decode(&d.config, nil, raws...)
+	if err != nil {
+		return err
+	}
+
+	var errs *packersdk.MultiError
+
+	if d.config.PersonalAccessToken == "" {
+		envToken := os.Getenv(helper.TokenEnvVar)
+		if envToken == "" {
+			errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(
```

Comment:
It looks like there's only one case that writes to `errs`. I'm admittedly not super familiar with Packer's SDK but would it make sense to just return the error directly here?

Reply:
Since this has been a pattern in various components of most packer plugins (e.g. [our builder](https://github.com/linode/packer-plugin-linode/blob/dev/builder/linode/config.go#L86)), I think it might be safer to keep it as the MultiError. And it will also be convenient for us to add more error in this `Configure` function in the future when we want to add more stuff that can return an error to it.

Reply:
Sounds good to me! Do you think it'd make sense to include the `config.Decode` error in the MultiError too?

Reply:
Actually I copied this pattern from the Amazon and other HashoCorp maintained plugins, lol

I think we can immediately return for config decoding error because no attribute in the config may be parsed later if the decoding failed.